hadoop-ozone/dist/src/main/k8s/examples/getting-started/test.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+export K8S_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+cd "$K8S_DIR"
+
+# shellcheck source=/dev/null
+source "../testlib.sh"
+
+rm -rf result
+
+regenerate_resources
+
+start_k8s_env
+
+execute_robot_test scm-0 smoketest/basic/basic.robot
+
+combine_reports 'getting-started'
+
+stop_k8s_env
+
+flekszible generate
```

Comment:
Here we have a main problem: some of the Kubernetes examples couldn't be started in the test environment. For example when the example is configured to schedule one datanode per real node, but we have only one node.

Therefore, before the tests the resource files are heavily modified:

 1. instead of using latest image, the current build is mounted to the `/opt/ozone` (similar to the docker tests)
 2. anti-affinity rules are removed (enable to run multiple datanode on the same node)
 3. real persistence is removed

These are executed by the `regenerate_resources` step at the beginning of the test. Which is something like this:

```
flekszible generate -t mount:hostPath="$OZONE_ROOT",path=/opt/hadoop -t image:image=apache/ozone-runner:20200420-1 -t ozone/onenode
```

As you can see we define three new transformations on the fly:

 1. `-t mount:hostPath="$OZONE_ROOT",path=/opt/hadoop` --> use the files from the current build
 2. `-t image:image=apache/ozone-runner:20200420-1`, use standard runner
 3.  `-t ozone/onenode` enable to schedule more datanode to the same node

The line which is commented by you:

```
flekszible generate
```

Is 100% optional. After the test, it restores the original state of the files. It can be added to the stop_k8s_env (as it can always be added). 

But based on the experience with the docker `test.sh` files, I would prefer to use more explicit lines in the test.sh.

As `test.sh` files are read frequently but modified only by a few times, I tend to make it slightly more verbose, but easier to understand (you can assume that `stop_k8s_env` stops kubernetes pods, but I wouldn't like to hide any hidden functionlaty there. 

But theses are just my thought, as it's a beginning of new test, I am really open to modify to any direction.

Reply:
Thanks for confirming my guess.

I'm fine with making `test.sh` more verbose at the cost of some duplication.  However, I propose extracting this optional `flekszible generate` to a function, and let it check if it's being run from `src` dir, to address my first point.

Reply:
> and let it check if it's being run from src dir, to address my first point

> I ran tests from target/..., had some errors and tried to find out what's wrong.

I am not sure what was the problem in your case. It supposed to work from the target directory. In fact `hadoop-ozone/dev-support/checks/kubernetes.sh` executes all the tests from target by default.

`apache/ozone:0.6.0-SNAPSHOT` is used because during the release it becomes `apache/ozone:0.6.0` which is the distributed container.

`./test.sh` (with the first flekszible generation) replaces all the images with a dev version (ozone-runner + mount).

It seems to be working for me and passed on the github CI, but if you see any error, please let me know as it should work everywhere (do you have the latest released flekszible?)

Reply:
Let me clarify: I think the script would have worked fine, but I tried to use k3d (k3s in docker) instead of plain k3s.  Since the local dir was not available in the container, it failed.

The confusing part was that the resource files did not mention volume mounts, since they were converted back at the end.  So only by looking at `regenerate_resources` in `testlib` did I realize what's wrong.

Reply:
Yes, that's what came into my mind, too.

Today, it's not possible to run test on remote clusters (k3d is a remote cluster) as local mount doesn't work.

I can introduce some options in the future (would be great to execute tests on real environments, too).

For now, I will add a log message to the `regenerate_resource` to print a warning to the stdout.